gin.go

```diff
@@ -194,6 +197,26 @@ func (engine *Engine) LoadHTMLFiles(files ...string) {
 	engine.SetHTMLTemplate(templ)
 }
 
+// LoadHTMLFilesRecursively loads recursively a slice of HTML files mathing an extension
+// and associates the result with HTML renderer.
+func (engine *Engine) LoadHTMLFilesRecursively(folder string, extensionsAllowed []string) {
+
+	files := []string{}
+	sort.Strings(extensionsAllowed)
+
+	filepath.Walk(folder, func(path string, info os.FileInfo, err error) error {
+		extension := strings.ToLower(filepath.Ext(path))
+		indexFound := sort.SearchStrings(extensionsAllowed, extension)
+		if indexFound < len(extensionsAllowed) && extensionsAllowed[indexFound] == extension {
+			files = append(files, path)
```

Comment:
There are have `sort` and `search` two steps.
why not use the follow code?

```go
for _, v := range extensionsAllowed {
    if strings.ToLower(v) == extension {
        files = append(files, path)
    }
}
```
moreover, `sort.SearchStrings` maybe have problem, see https://github.com/golang/go/issues/25382